crates/sui-adapter/src/adapter.rs

```diff
@@ -113,30 +111,21 @@ pub fn new_move_vm(
     .map_err(|_| SuiError::ExecutionInvariantViolation)
 }
 
-pub fn new_session<
-    'v,
-    'r,
-    E: Debug,
-    S: ResourceResolver<Error = E>
-        + ModuleResolver<Error = E>
-        + LinkageResolver<Error = E>
-        + ChildObjectResolver,
->(
-    vm: &'v MoveVM,
-    state_view: &'r S,
+pub fn new_native_extensions<'r>(
```

Comment:
Originally it was because we were going to create sessions multiple times but against the same native context extensions. We don't do that anymore, but the fact that the session's store is a `LinkageView` now (which is a programmable-transaction-specific type) makes it convenient to do that separately anyway.